directory/assets/configuration/spec.yaml

```diff
@@ -0,0 +1,101 @@
+      value:
+        type: string
+        example: <DIRECTORY_TAG_KEY>
+    - name: filetagname
+      description: |
+        The name of the key for the tag used for each file, the value is the filename.
+        The resulting tag is "<filetagname>:<filename>". The option is only used when `filegauges` is enabled
+        and if the directory contains 20 files or less (including sub-directories).
+      value:
+        type: string
+        example: <TAG_KEY_FILENAME>
+    - name: filegauges
+      description: When enabled, stats are an individual gauge per file (max. 20 files!) and not a histogram of the whole directory.
+      value:
+        type: boolean
+        example: false
+    - name: pattern
+      description: |
+        The `fnmatch` pattern to use when reading the `directory`'s files.
+        The pattern is matched against the files' absolute paths and relative paths in `directory`.
+      value:
+        type: string
+        example: '*'
+    - name: exclude_dirs
+      description: A list fo regular expressions used to blacklist directory names. For example, node_modules or vendor.
+      value:
+        type: array
+        items:
+          type: string
+        example:
+          - ^__pycache__$
+          - ^build$
+          - ^node_modules$
+          - ^vendor$
+    - name: dirs_patterns_full
+      description: |
+        Make exclude_dirs patterns operate on the full directory path (may be slightly slower).
+        When enabled, this option allows you to specify the absolute path.
+        This ensures multiple directories with the same name aren't all excluded.
+
+        Otherwise, this option excludes any directory in the traversal with the provided name.
+      value:
+        type: boolean
+        example: false
+    - name: recursive
+      description: When enabled, the stats recurse into directories.
+      value:
+        type: boolean
+        example: false
+    - name: follow_symlinks
+      description: |
+        Follow symlinks when traversing directories.
+        When the option `recursive` is enabled, setting this option to `false` helps avoid symlink loops.
+      value:
+        type: boolean
+        example: true
+    - name: stat_follow_symlinks
+      description: |
+        When enabled, the stats are gathered for the files the symlinks are pointing to.
+        Otherwise, the stats are gathered for the symlinks themselves.
+      value:
+        type: boolean
+        example: true
+    - name: countonly
+      description: When enabled, the stats only count the number of files matching the pattern.
+      value:
+        type: boolean
+        example: false
+    - name: ignore_missing
+      description: When enabled, the check does not raise an exception on missing or inaccessible directories.
+      value:
+        type: boolean
+        example: false
+    - template: init_config/default
+  - template: instances
+    options:
+    - template: instances/default
```

Comment:
all the options are in init instead of instances. Is this right?

Reply:
My bad, the init_config is empty. Misread the yaml sections 😅 . I've moved the options to instances